addons/nuxeo-retention/nuxeo-retention-core/src/main/java/org/nuxeo/retention/adapters/RetentionRule.java

```diff
@@ -0,0 +1,246 @@
+        return (Long) document.getPropertyValue(RetentionConstants.DURATION_MONTHS_PROP);
+    }
+
+    public Long getDurationYears() {
+        return (Long) document.getPropertyValue(RetentionConstants.DURATION_YEARS_PROP);
+    }
+
+    public List<String> getEndActions() {
+        Serializable propertyValue = document.getPropertyValue(RetentionConstants.END_ACTIONS_PROP);
+        if (propertyValue == null) {
+            return Collections.emptyList();
+        }
+        return Arrays.asList((String[]) propertyValue);
+    }
+
+    public String getExpression() {
+        return (String) document.getPropertyValue(RetentionConstants.EXPRESSION_PROP);
+    }
+
+    public String getMetadataXpath() {
+        return (String) document.getPropertyValue(RetentionConstants.METADATA_XPATH_PROP);
+    }
+
+    public Calendar getRetainUntilDateFrom(Calendar calendar) {
+        LocalDateTime datetime = LocalDateTime.ofInstant(calendar.getTime().toInstant(), ZoneId.systemDefault());
+        return getRetainUntilDateFrom(datetime);
+    }
+
+    protected Calendar getRetainUntilDateFrom(LocalDateTime datetime) {
+        LocalDateTime localDateTime = datetime.plusYears(getDurationYears())
+                                              .plusMonths(getDurationMonths())
+                                              .plusDays(getDurationDays())
+                                              .plusNanos(getDurationMillis() * 1000000);
+        return GregorianCalendar.from(localDateTime.atZone(ZoneId.systemDefault()));
+    }
+
+    public Calendar getRetainUntilDateFromNow() {
+        return getRetainUntilDateFrom(LocalDateTime.now());
+    }
+
+    public String getStartingPointEvent() {
+        return (String) document.getPropertyValue(RetentionConstants.STARTING_POINT_EVENT_PROP);
+    }
+
+    public String getStartingPointExpression() {
+        return (String) document.getPropertyValue(RetentionConstants.STARTING_POINT_EXPRESSION_PROP);
+    }
+
+    public StartingPointPolicy getStartingPointPolicy() {
+        String value = (String) document.getPropertyValue(RetentionConstants.STARTING_POINT_POLICY_PROP);
+        if (value != null) {
+            return StartingPointPolicy.valueOf(value.toUpperCase());
+        }
+        return null;
+
+    }
+
+    public boolean isAfterDely() {
+        return StartingPointPolicy.AFTER_DELAY.equals(getStartingPointPolicy());
+    }
+
+    public boolean isAuto() {
+        return ApplicationPolicy.AUTO.name().toLowerCase().equals(getApplicationPolicy());
+    }
+
+    public boolean isDocTypeAccepted(String docType) {
+        List<String> types = getDocTypes();
+        return types == null || types.isEmpty() || types.contains(docType);
+    }
+
+    public boolean isEnabled() {
+        return (boolean) document.getPropertyValue(RetentionConstants.ENABLED_PROP);
+    }
+
+    public boolean isEventBased() {
+        return StartingPointPolicy.EVENT_BASED.equals(getStartingPointPolicy());
```

Comment:
as `getStartingPointPolicy` returns the `enum` we can use directly the `==` operator instead of `equals` and i think `getStartingPointPolicy` can be adapted to be part of the `enum` itself  and i will see the enum improved with:

```
    public enum StartingPointPolicy {
        IMMEDIATE, AFTER_DELAY, EVENT_BASED, METADATA_BASED;

        public static StartingPointPolicy fromValue(String value) {
            if (value != null) {
                return StartingPointPolicy.valueOf(value.toUpperCase());
            }
            return null;
        }

        public static StartingPointPolicy fromDocument(DocumentModel document) {
            if (document != null) {
                return fromValue((String) document.getPropertyValue(RetentionConstants.STARTING_POINT_POLICY_PROP));
            }
            return null;
        }
    }
```

and the call 
```
public boolean isAfterDely() {
        return AFTER_DELAY == StartingPointPolicy.fromDocument(document);
    }
```

Reply:
Ok for `==`, I'll refactor the upper/lower stuff later it's rather ugly.